### PR TITLE
Inserter: Update the search form placeholder text when changing tabs

### PR DESCRIPTION
**Pull request by apeatling**
## Description
When changing tabs between: "Blocks | Patterns | Reusable", the search form does not change its placeholder text. This means when looking at the Patterns or Reusable tabs, it still says "Search for a block".

This PR makes sure that the placeholder in the search input field changes depending on the tab you have selected.

## How has this been tested?
Tested locally and all tests run and passed.

## Types of changes
Bug fix, non breaking change.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by shaunandrews — APPROVED** [review]
Does what it says.

File: packages/block-editor/src/components/inserter/menu.js [hunk 1]
```diff
@@ -112,6 +113,16 @@ function InserterMenu( {
 		/>
 	);
 
+	const searchFormPlaceholder = () => {
+		if ( activeTab === 'blocks' ) {
```
[the remark this answers is not in the document]
Comment (apeatling) [reply packages/block-editor/src/components/inserter/menu.js]: 👍

**Comment by afercia** [comment]
Seems to me this change made the "quick inserter" search input unlabelled (and with no placeholder text): 

Screenshot 2020-08-25 at 14 26 53

- there's a properly associated and visually hidden `<label>` element but it's now empty
- the search input has no placeholder text

Seems to me the new function `searchFormPlaceholder()` works for the main inserted but now the `placeholder` prop for the quick inserter is not set any longer.

@apeatling can you please have a look, when you have a chance?

Aside: as mentioned in #24494 and in other previous conversations, it would be really great to have some kind of automated tests to make sure all form elements are properly labelled, see als #6098. /Cc @youknowriad

**Comment by apeatling** [comment]
Will get this fixed this morning. 👍

**Pull request merged**